src/github.com/mozilla-services/pushgo/simplepush/aws.go

```diff
@@ -34,11 +58,21 @@ func GetAWSPublicHostname() (hostname string, err error) {
 		err = fmt.Errorf("Unexpected status code: %d", resp.StatusCode)
 		return
 	}
-	hostBytes, err := ioutil.ReadAll(resp.Body)
+	respBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
ReadAll doesn't seem to close the resp.Body, the http package says resp.Body must always be closed after reading it.